src/gui/torrentcontentmodel.cpp

```diff
@@ -81,44 +86,80 @@ namespace
             return cached;
         }
     };
-#if defined(Q_OS_WIN)
-    // See QTBUG-25319 for explanation why this is required
-    class WinShellFileIconProvider: public UnifiedFileIconProvider
+
+#ifdef QBT_PIXMAP_CACHE_FOR_FILE_ICONS
+    struct PixmapCacheSetup
+    {
+        static const int PixmapCacheForIconsSize = 2 * 1024 * 1024; // 2 MiB for file icons
+
+        PixmapCacheSetup()
+        {
+            QPixmapCache::setCacheLimit(QPixmapCache::cacheLimit() + PixmapCacheForIconsSize);
+        }
+
+        ~PixmapCacheSetup()
+        {
+            Q_ASSERT(QPixmapCache::cacheLimit() > PixmapCacheForIconsSize);
+            QPixmapCache::setCacheLimit(QPixmapCache::cacheLimit() - PixmapCacheForIconsSize);
+        }
+    };
+
+    PixmapCacheSetup pixmapCacheSetup;
```

Comment:
Since there is non-trivial constructor and destructor, a reliable compiler has to understand what is going on there. I've added `Q_DECL_UNUSED` to the class declaration through.